app/main.py

```diff
@@ -1,2 +1,131 @@
+
+from typing import List, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, elements: List[tuple] = None) -> None:
+        if elements is None:
+            elements = []
+        self._bucket_size = 8
+        self._bucket_resize = self._bucket_size * 2 / 3
+        while len(elements) > self._bucket_resize:
+            self._bucket_size *= 2
+            self._bucket_resize = self._bucket_size * 2 / 3
+        self._buckets = [[] for i in range(self._bucket_size)]
+        if len(elements):
+            self._assign_buckets(elements)
+
+    def _assign_buckets(self, elements: List[tuple]) -> None:
+        for _key, _value in elements:
+            if isinstance(_key, list | dict | set):
+                raise KeyError(
+                    f'The key can`t be like this: "{_key}" is "{type(_key)}"'
+                )
+            hashed_value = hash(_key)
+            index = hashed_value % self._bucket_size
+
+            while self._buckets[index]:
+                if self._buckets[index][0] == _key:
+                    break
+                index = (index + 1) % self._bucket_size
+
+            self._buckets[index] = (_key, _value,)
+
+    def _get_buckets_full(self) -> list[Any]:
+        return [
+            full_bucket for full_bucket in self._buckets
+            if full_bucket and full_bucket[1] != "remote"
+        ]
+
+    def _resize(
+            self,
+            new_element: List[tuple],
+            len_new_elements: int = 0
+    ) -> None:
+        current_buckets = self._get_buckets_full()
+        current_buckets += new_element
+
+        while len(self) + len_new_elements > self._bucket_resize:
```

Comment:
`len(self)` looks a bit strange. Does that make sense to set elements as an attribute of instance?

Reply:
I don't understand what exactly needs to be done

Reply:
Create class attribute as `self.lenght=0` and than, after added or removed element from Dictionary change this attr